### PR TITLE
Remove deprecated LegacyMap

### LINKED ISSUE #2598: Remove usage od deprectaed `LegacyMap`
### Which components does the task require to be changed? (think hard pls)

snforge

### Description

In some places in packages in `crates/forge/tests/data` we're still using deprecated `LegacyMap`. We should update it to use `starknet::storage::Map` instead.

Example usages:
https://github.com/foundry-rs/starknet-foundry/blob/8df49f64d420740e1b946f68ec7c414af77aca0e/crates/sncast/tests/data/scripts/map_script/contracts/src/lib.cairo#L13

https://github.com/foundry-rs/starknet-foundry/blob/8df49f64d420740e1b946f68ec7c414af77aca0e/crates/forge/tests/data/component_macros/src/oz_ac_component.cairo#L18

### PR DESCRIPTION
**Pull request by sprtd**
Closes #2598

## Introduced changes

- Replace `LegacyMap` storage mapping with `Map`

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/README.md
Comment (franciszekjob) [inline crates/sncast/README.md:1]: Changing readme is not needed.

File: design_documents/stark_curve_signatures.md
Comment (franciszekjob) [inline design_documents/stark_curve_signatures.md:1]: This file shouldn't be changed as well.

File: design_documents/store_load_cheatcodes.md
Comment (franciszekjob) [inline design_documents/store_load_cheatcodes.md:1]: These changes are not needed as well.

File: crates/forge/tests/integration/test_state.rs [hunk 1]
```diff
@@ -856,6 +856,5 @@ fn felt252_dict_usage() {
     ));
 
     let result = run_test_case(&test);
-
```
Comment (franciszekjob) [inline crates/forge/tests/integration/test_state.rs:859]: No need to remove this line.

**Review by franciszekjob — CHANGES_REQUESTED** [review]
Please fix failing tests and formatting and we should be good to go 👍 .

**Review by franciszekjob — CHANGES_REQUESTED** [review]
Please fix failing tests and linting 🙏 .

**Comment by sprtd** [comment]
@franciszekjob 
these are the failing e2e tests:

```
test e2e::running::detailed_resources_flag ... FAILED
test e2e::running::simple_package_with_git_dependency ... FAILED
test e2e::running::storage_access_default_values ... FAILED 
```

**Comment by franciszekjob** [comment]
Hi @sprtd, I see the tests are still failing. Did you try to fix them?

**Comment by sprtd** [comment]
> Hi @sprtd, I see the tests are still failing. Did you try to fix them?

fixed the rest. 
however, `cargo test detailed_resources_flag` fails

**Review by franciszekjob — APPROVED** [review]
LGTM :)

**Pull request merged**